validators/code.validator.js

```diff
@@ -1,55 +1,69 @@
 const Joi = require('joi')
-const { PROMPTV1, PROMPTV2, MULTIFILE } = require('../enums/supportedLanguages')
+const { PROMPTV1, PROMPTV2, MULTIFILE, PROMPTV3 } = require('../enums/supportedLanguages')
 const { FRONTEND_REACT_JASMINE, FRONTEND_STATIC_JASMINE } = require('../enums/supportedMultifileSetupTypes')
 
-const isValidForExecute = async (body) => {
-    const schema = Joi.object({
+const _getAiLanguageSchema = () => {
+    return Joi.object({
         language: Joi.string().required(),
-        question: Joi.string().when('language', {
-            is: [PROMPTV1, PROMPTV2],
-            then: Joi.required(),
-            otherwise: Joi.optional(),
-        }),
-        userAnswer: Joi.string().when('language', {
-            is: [PROMPTV1, PROMPTV2],
-            then: Joi.required(),
-            otherwise: Joi.forbidden(),
-        }),
-        rubric: Joi.string().when('language', {
-            is: [PROMPTV1, PROMPTV2],
-            then: Joi.optional(),
-            otherwise: Joi.forbidden(),
-        }),
-        script: Joi.string()
-            .when('language', {
-                is: [PROMPTV1, PROMPTV2, MULTIFILE],
-                then: Joi.optional(),
-                otherwise: Joi.required(),
-            }),
-        url: Joi.string().trim()
-            .when('language', {
-                is: MULTIFILE,
-                then: Joi.required(),
-                otherwise: Joi.forbidden(),
-            }),
-        type: Joi.string().trim()
+        question: Joi.string().required(),
+        userAnswer: Joi.string().required(),
+        rubric: Joi.string().optional(),
+        script: Joi.string().optional(),
+        points: Joi.number().integer().optional(),
+        hasInputFiles: Joi.bool(),
+        args: Joi.string(),
+        stdin: Joi.string(),
+    })
+}
+
+const _getMultiFileLanguageSchema = () => {
+    return Joi.object({
+        language: Joi.string().required(),
+        question: Joi.string().optional(),
+        script: Joi.string().optional(),
+        url: Joi.string().trim().required(),
+        type: Joi.string()
+            .trim()
             .valid(FRONTEND_REACT_JASMINE, FRONTEND_STATIC_JASMINE)
-            .when('language', {
-                is: MULTIFILE,
-                then: Joi.required(),
-                otherwise: Joi.forbidden(),
-            }),
-        non_editable_files: Joi.object().pattern(
-            Joi.string(),
-            Joi.string().pattern(/^[a-fA-F0-9]{64}$/)
-        ).optional(),
-        points: Joi.number().integer().optional(), // totalScore
+            .required(),
+        non_editable_files: Joi.object()
+            .pattern(Joi.string(), Joi.string().pattern(/^[a-fA-F0-9]{64}$/))
+            .optional(),
+        points: Joi.number().integer().optional(),
         hasInputFiles: Joi.bool(),
         args: Joi.string(),
         stdin: Joi.string(),
     })
```

Comment:
See if we can have a base schema for repeated values and then extend the individual validations on top of the base schema.

Reply:
I believe doing this will create confusion as now whoever is contributing would need to make sure their flow and the common schema flow works correctly. With the current approach they would just need to verify their own flow